### ISSUE TITLE
The message for files starting with a dot is incorrect

### ISSUE THREAD

**Issue by cpitclaudel** [eslint/eslint#3556]
Eslint is hardcoded to not process files that start with a dot, but claims that it did so because of `.eslintignore`. This behaviour is due to https://github.com/eslint/eslint/blob/3de24f103301138171094333d89be51f30839c87/lib/util/traverse.js#L44 and https://github.com/eslint/eslint/blob/3de24f103301138171094333d89be51f30839c87/lib/cli-engine.js#L398 (files excluded by the first link will end up not being in the `processed` list of the second link, which in turn will cause them to be reported as excluded by `eslintignore`).

Ideally eslintrc could process files starting with a dot and explicitly passed on the command line; if not, the error message should at least probably be changed.

Here's how to reproduce this:

```sh
$ cat .test.js 
var foo = bar;
$ cat .eslintrc 
extends:
    "eslint:recommended"
$ eslint --format\=checkstyle .test.js
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name=".test.js"><error line="undefined" column="undefined" severity="warning" message="File ignored because of your .eslintignore file. Use --no-ignore to override." source="" /></file></checkstyle>
$ cat .eslintignore
cat: .eslintignore: No such file or directory
$ eslint --format\=checkstyle --no-ignore .test.js
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"></checkstyle>
```

This issue is connected to https://github.com/flycheck/flycheck/issues/735

Cheers,
Clément.

**Comment by BYK** [comment]
This looks a lot like a bug. Gonna investigate.

**Comment by BYK** [comment]
Looks like this is a side effect from the fix for #1077 (and #814). I'd argue "dot skipping" should apply to folder but not files, especially if they are explicitly passed. What do you think @nzakas ?

Also, #3496 may possibly fix this.

**Comment by nzakas** [comment]
Yeah, we should lint these files if passed directly on the command line, but should not if that are found during a normal traversal.

**Comment by nzakas** [comment]
@byk still working on this?

**Comment by BYK** [comment]
@nzakas nope, sorry. Fell off my radar. I'll unassign from myself but will try to create time for it soon.

**Comment by gyandeeps** [comment]
Closing as confirmed with current master that if you pass in a .file directly it will lint it.
